compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -561,6 +561,13 @@ object Checking {
         fail(CannotHaveSameNameAs(sym, cls, CannotHaveSameNameAs.CannotBeOverridden))
         sym.setFlag(Private) // break the overriding relationship by making sym Private
       }
+    if sym.isWrappedToplevelDef && !sym.isType && sym.flags.is(Infix, butNot = Extension) then
+      val defKind = 
```

Comment:
IMO this kind of sticks out like a sore thumb, but adding this logic to `ModifierNotAllowedForDefinition` isn't really possible since it'd make it specific to this one error. Maybe just subtype `ModifierNotAllowedForDefinition` instead?

Reply:
Finding the keyword from the flags seems like a common operation, are you sure there isn't already a method doing this ?

Reply:
Best I found was https://github.com/lampepfl/dotty/blob/ed319e8a41ef46ea5d7db35f3608eb17d8bbe7d3/compiler/src/dotty/tools/dotc/reporting/messages.scala#L2617 (I also kindly stole the name) which also inlines this logic (albeit in a more tolerable manner)

Reply:
Hmm I see, then maybe you should create such a method ?
Not an expert on the design of the compiler tho @mbovel what are your thoughts ?

Reply:
Yeah, that'd make the most sense probably. Where do helpers like these live?

Reply:
There was a file with some extension methods on `Symbol`, but it might be ScalaDoc specific
I would advise you find the place were resides the most similar methods, and stick it there ^^'

Reply:
See my suggestion in my new review: https://github.com/lampepfl/dotty/pull/17994#discussion_r1261142044.